pkg/controller/controlplane/valuesprovider.go

```diff
@@ -341,9 +342,41 @@ func (vp *valuesProvider) GetConfigChartValues(
 	return getConfigChartValues(infraStatus, cp, cluster, auth)
 }
 
+func deleteLegacyCloudProviderConfigMap(ctx context.Context, c client.Client, namespace string) error {
+	ccmDeployment := &appsv1.Deployment{}
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      azure.CloudProviderConfigName,
+			Namespace: namespace,
+		},
+	}
+	if err := c.Get(ctx, kutil.Key(namespace, azure.CloudControllerManagerName), ccmDeployment); client.IgnoreNotFound(err) != nil {
```

Comment:
Is this really necessary, i.e., can't we simply always try to delete the config map and ignore the 'not found' error, similar to your `deleteLegacyCloudProviderDiskConfigMap` function? We don't really safe anything by first reading the CCM deployment, do we?
OTOH, I don't have any strong opinion, I didn't want to comment at all, but then I saw the `deleteLegacyCloudProviderDiskConfigMap` implementation and was wondering why there isn't similar code that determines whether the config map should be deleted.

Reply:
Okay, I agree that this check doesn't make too much sense.